Frameworks/CoreText/CTLine.mm

```diff
@@ -56,64 +120,95 @@ static CFStringRef __CTCreateReversedString(CFStringRef string) {
     return CFStringCreateWithCharactersNoCopy(kCFAllocatorDefault, characters.release(), bufLen, nullptr);
 }
 
-static NSMutableAttributedString* _getTruncatedStringFromSourceLine(CTLineRef line,
-                                                                    CTLineTruncationType truncationType,
-                                                                    double widthToExtract);
-
-@implementation _CTLine : NSObject
-- (instancetype)init {
-    if (self = [super init]) {
-        _runs.attach(CFArrayCreateMutable(kCFAllocatorDefault, 0, &kCFTypeArrayCallBacks));
-    }
-    return self;
+CTLineRef _CTLineCreate() {
+    return __CTLine::CreateInstance(kCFAllocatorDefault);
 }
 
-- (instancetype)copyWithZone:(NSZone*)zone {
-    _CTLine* ret = [_CTLine new];
-    ret->_strRange = _strRange;
-    ret->_width = _width;
-    ret->_ascent = _ascent;
-    ret->_descent = _descent;
-    ret->_leading = _leading;
-    ret->_glyphCount = _glyphCount;
-    ret->_runs.attach(CFArrayCreateMutableCopy(kCFAllocatorDefault, 0, _runs));
-    ret->_relativeXOffset = _relativeXOffset;
+static CFMutableAttributedStringRef __getTruncatedStringFromSourceLine(CTLineRef sourceLine,
+                                                                       CTLineTruncationType truncationType,
+                                                                       double widthToExtract) {
+    CFArrayRef glyphRuns = CTLineGetGlyphRuns(sourceLine);
+    CFIndex numberOfRuns = CFArrayGetCount(glyphRuns);
+    double extractedWidth = 0;
+    bool done = 0;
```

Comment:
LOL this is from earlier code.
I'll update it.

---
In reply to: [112725988](https://github.com/Microsoft/WinObjC/pull/2586#discussion_r112725988) [](ancestors = 112725988)